app/cdap/components/shared/ConfigurationGroup/utilities/index.ts

```diff
@@ -382,17 +382,18 @@ export function countErrors(
  *
  * @param values latest field values
  * @param filteredConfigurationGroups configuration-groups from Widget JSON
- * @param addDefaults add default values for visible fields that are empty
+ * @param defaultValues default field values
  */
-export function removeFilteredProperties(values, filteredConfigurationGroups, addDefaults = false) {
+export function removeFilteredProperties(values, filteredConfigurationGroups, defaultValues = {}) {
```

Comment:
Nit: Consider `baseValues` instead of `defaultValues`? Since they're not always the "defaults" as they were before.